api/src/client/dynamics/DynamicsLicenseApplicationIdClient.test.ts

```diff
@@ -0,0 +1,235 @@
+          rgb_number: "39PW00105401",
+          rgb_startdate: null,
+          rgb_versioncode: 100000000,
+          rgb_expirationdate: null,
+          statecode: 1,
+          statuscode: 100000005,
+          rgb_applicationid: "license-id-1357",
+        },
+        {
+          rgb_appnumber: "TELI00030500",
+          rgb_number: "39PW00105400",
+          rgb_startdate: null,
+          rgb_versioncode: 100000000,
+          rgb_expirationdate: null,
+          statecode: 0,
+          statuscode: 100000005,
+          rgb_applicationid: "license-id-1357",
+        },
+      ],
+    };
+
+    mockAxios.get.mockResolvedValue({ data: mockLicenseApplicationIdResponse });
+    expect(
+      await client.getLicenseApplicationId(mockAccessToken, mockBusinessId, "Public Movers and Warehousemen")
+    ).toEqual({
+      expirationDate: "",
+      applicationId: "license-id-1357",
+      licenseStatus: "APPROVED",
+    });
+  });
+
+  it("filters out applications whose rgb_number doesn't end with 00 and retains the main applications", async () => {
+    const mockLicenseApplicationIdResponse = {
+      value: [
+        {
+          rgb_appnumber: "WLI00030610",
+          rgb_number: "39PW00105401",
+          rgb_startdate: "2023-06-08T04:44:34Z",
+          rgb_versioncode: 100000000,
+          rgb_expirationdate: "2023-09-30T00:00:00Z",
+          statecode: 0,
+          statuscode: 100000006,
+          rgb_applicationid: "8b7bbfb9-b605-ee11-a81c-001dd80648b3",
+          _rgb_businessid_value: "6182f96b-2105-ee11-a81c-001dd805ef6c",
+        },
+        {
+          rgb_appnumber: "MLI00030615",
+          rgb_number: "39PM00105502",
+          rgb_startdate: "2023-06-08T04:52:25Z",
+          rgb_versioncode: 100000000,
+          rgb_expirationdate: "2023-09-30T00:00:00Z",
+          statecode: 0,
+          statuscode: 100000006,
+          rgb_applicationid: "1abb2efc-b705-ee11-a81c-001dd80648b3",
+          _rgb_businessid_value: "6182f96b-2105-ee11-a81c-001dd805ef6c",
+        },
+        {
+          rgb_appnumber: "MLI00030602",
+          rgb_number: "39PM00105300",
+          rgb_startdate: "2023-06-07T11:49:24Z",
+          rgb_versioncode: 100000000,
+          rgb_expirationdate: "2023-09-30T00:00:00Z",
+          statecode: 0,
+          statuscode: 100000006,
+          rgb_applicationid: "7b5c178c-2805-ee11-a81c-001dd805ef6c",
+          _rgb_businessid_value: "6182f96b-2105-ee11-a81c-001dd805ef6c",
+        },
+      ],
+    };
+
+    mockAxios.get.mockResolvedValue({ data: mockLicenseApplicationIdResponse });
+    expect(
+      await client.getLicenseApplicationId(mockAccessToken, mockBusinessId, "Public Movers and Warehousemen")
+    ).toEqual({
+      expirationDate: "2023-09-30T00:00:00Z",
+      applicationId: "7b5c178c-2805-ee11-a81c-001dd805ef6c",
+      licenseStatus: "ACTIVE",
+    });
+  });
+
+  it("throws NO_MAIN_APPS error if there are no matching main applications", async () => {
```

Comment:
[sand] I think it'd be helpful for this test name (and the following one) to state/clarify what makes something a main application (it seems like ending with "00"?)

Like, just from reading this test setup, I can't immediately tell  _why_ this particular setup has "no main apps" unless you already know.